Dynatrace/dynatrace-operator
# Improve reconcile flow for apimonitoring

## Description

The reconcile flow in the apimonitoring reconciler is/was flawed.
- It required multiple runs to set up everything

Problem:
- `CreateOrUpdateKubernetesSetting` will:
  - Create a `Monitored Entity` if no scope was given (scope == monitored entity ID), this new `Monitored Entity` will be used as the scope for the created `Settings` object
  - Create/Update a `Settings` object with scope

So to not have to run the whole api-monitoring reconciler to set the monitored entity ID in the dynakube status, we should run the monitored-enitity-reconciler before and after `CreateOrUpdateKubernetesSetting`
- "before" is needed so we use the correct scope for creation/update if it exists
- "after" is needed so we actually set/know the monitored entity ID in the dynakube, so other features can use it.

Also moved the Kubernetes-app related logic to be the last thing, as that also relies on the fact that the `Monitored Entity` AND`Settings` object exists 

## How can this be tested?
When your tenant doesn't have a `Monitored Entity` for a given cluster, apply a dynakube with an `kubernetes-monitoring` ActiveGate, and see that `dk.Status.KubernetesClusterMEID` is set without having to wait for a while.
